cub/test/catch2_test_device_reduce_vsmem.cu

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "insert_nested_NVTX_range_guard.h"
+
+#include <cub/device/device_reduce.cuh>
+
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_launch_helper.h"
+#include <c2h/catch2_test_helper.h>
+#include <c2h/custom_type.h>
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Reduce, device_reduce);
+
+// huge_data<x> -> sizeof(AgentReduce<...>::TempStorage) on sm86
+// 16 -> 224
+// 32 -> 368
+// 64 -> 656
+// 4Ki -> 8216
+// 16Ki -> 32792
+// 24Ki -> 49176
+// 32Ki -> 65560 (but causes Error: Formal parameter space overflowed)
+using value_types =
+  c2h::type_list<c2h::custom_type_t<c2h::equal_comparable_t, c2h::accumulateable_t, c2h::huge_data<24 * 1024>::type>>;
```

Comment:
Compile times will be interesting

Reply:
Even better, [one CI run](https://github.com/NVIDIA/cccl/actions/runs/18075194835/job/51430644070?pr=6050#step:4:7722) fails with:
```
  /home/coder/cccl/lib/cmake/cub/../../../cub/cub/thread/thread_load.cuh:123:48: fatal error: instantiating fold expression with 3073 arguments exceeded expression nesting limit of 256
    123 |   ((dst[Is] = ThreadLoad<MODIFIER>(src + Is)), ...);
        |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
```

Reply:
Another [one](https://github.com/NVIDIA/cccl/actions/runs/18075194835/job/51430644288?pr=6050#step:4:1216):
```
reduce.cuh(243): Error: Formal parameter space overflowed (24614 bytes required, max 4096 bytes allowed)
```